### PR TITLE
Improve all rule checking slots with standard BlockError representation

### PR DESCRIPTION
**Pull request by sczyh30**
Signed-off-by: Eric Zhao <sczyh16@gmail.com>

### Describe what this PR does / why we need it

Improve all rule checking slots with standard `BlockError` representation.

### Does this pull request fix one issue?

NONE

### Describe how you did it

This PR follows a simple BlockError specification:

- The `Error()` SHOULD start with `SentinelBlockError: ${blockType}`. If block message is provided, append it to the `Error()` value. The `blockMsg` is not required.
- It's recommended to carry the triggered rule and the "snapshot" value in the `BlockError`.

This PR improves all rule checking slots with more standard `BlockError` representation.

### Describe how to verify it

Run the test cases.

### Special notes for reviews

NONE

### CODE REVIEW

File: core/hotspot/traffic_shaping.go [hunk 1]
```diff
@@ -87,16 +87,14 @@ func (c *baseTrafficShapingController) performCheckingForConcurrencyMetric(arg i
 			return nil
 		}
 		return base.NewTokenResultBlockedWithCause(base.BlockTypeHotSpotParamFlow,
-			fmt.Sprintf("Blocked by specific concurrency threshold, arg: %+v, current concurrency: %d,  specific concurrency: %d", arg, concurrency, specificConcurrency),
-			c.BoundRule(), concurrency)
+			fmt.Sprintf("arg=%v", arg), c.BoundRule(), nil)
```
Comment (louyuting) [inline core/hotspot/traffic_shaping.go:90]: Maybe we should carry the snapshot?

File: core/flow/tc_throttling.go [hunk 2]
```diff
@@ -34,7 +34,7 @@ func (c *ThrottlingChecker) DoCheck(_ base.StatNode, acquireCount uint32, thresh
 		return nil
 	}
 	if threshold <= 0 {
-		return base.NewTokenResultBlocked(base.BlockTypeFlow, "Flow")
+		return base.NewTokenResultBlocked(base.BlockTypeFlow)
```
Comment (louyuting) [inline core/flow/tc_throttling.go:37]: Maybe we need to carry the triggered rule?

File: core/hotspot/traffic_shaping.go (hunk 1 above)
Comment (sczyh30) [reply core/hotspot/traffic_shaping.go]: Updated.

File: core/flow/tc_throttling.go (hunk 2 above)
Comment (sczyh30) [reply core/flow/tc_throttling.go:37]: Yes, the design regarding tc_throttling.go should be improved.

**Pull request merged**